packages/fxa-auth-server/lib/routes/account.ts

```diff
@@ -1011,50 +1011,48 @@ export class AccountHandler {
         verificationMethod
       );
 
-      // For new sync logins that don't send some other sort of email,
+      // For new logins that don't send some other sort of email,
       // send an after-the-fact notification email so that the user
       // is aware that something happened on their account.
       if (accountRecord.primaryEmail.isVerified) {
         if (sessionToken.tokenVerified || !sessionToken.mustVerify) {
-          if (requestHelper.wantsKeys(request)) {
-            const geoData = request.app.geo;
-            const service =
-              (request.payload as any).service || request.query.service;
-            const ip = request.app.clientAddress;
-            const { deviceId, flowId, flowBeginTime } = await request.app
-              .metricsContext;
-
-            try {
-              await this.mailer.sendNewDeviceLoginEmail(
-                accountRecord.emails,
-                accountRecord,
-                {
-                  acceptLanguage: request.app.acceptLanguage,
-                  deviceId,
-                  flowId,
-                  flowBeginTime,
-                  ip,
-                  location: geoData.location,
-                  service,
-                  timeZone: geoData.timeZone,
-                  uaBrowser: request.app.ua.browser,
-                  uaBrowserVersion: request.app.ua.browserVersion,
-                  uaOS: request.app.ua.os,
-                  uaOSVersion: request.app.ua.osVersion,
-                  uaDeviceType: request.app.ua.deviceType,
-                  uid: sessionToken.uid,
-                }
-              );
-            } catch (err) {
-              // If we couldn't email them, no big deal. Log
-              // and pretend everything worked.
-              this.log.trace(
-                'Account.login.sendNewDeviceLoginNotification.error',
-                {
-                  error: err,
-                }
-              );
-            }
+          const geoData = request.app.geo;
+          const service =
+           (request.payload as any).service || request.query.service;
```

Comment:
Previously when requestHelper.wantsKey was in use, it would ensure that request.query wasn't undefined/null. Perhaps request.query still needs to be checked?

Reply:
I think this is probably ok since it goes through joi validation.